### PR TITLE
Vendor gardener/gardener@v1.50.1

### PR DESCRIPTION
**Pull request by bd3lage**
**How to categorize this PR?**

/area open-source
/kind enhancement
/platform gcp

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
The following dependency is updated:
- github.com/gardener/gardener: v1.48.0 -> v1.50.1
```

### CODE REVIEW

**Comment by gardener-robot-ci-1** [comment]
Thank you @bd3lage for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

**Review by kon-angelo — DISMISSED** [review]
/lgtm

**Review by acumino — COMMENTED** [review]
Please also adapt the changes similar to this https://github.com/gardener/gardener-extension-provider-aws/pull/570

**Comment by bd3lage** [comment]
> Please also adapt the changes similar to this [gardener/gardener-extension-provider-aws#570](https://github.com/gardener/gardener-extension-provider-aws/pull/570)

@acumino You mean the [make metrics bind address configurable](https://github.com/gardener/gardener-extension-provider-aws/pull/570/commits/f4847ad706c4c88c1c18f367139560985d3067ba) part, right ?

File: go.mod [hunk 1]
```diff
@@ -138,34 +139,34 @@ require (
...
-	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.23.3
-	k8s.io/apimachinery => k8s.io/apimachinery v0.23.3
-	k8s.io/apiserver => k8s.io/apiserver v0.23.3
+	github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.12.1 // keep this value in sync with sigs.k8s.io/controller-runtime
+	k8s.io/api => k8s.io/api v0.24.2
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.24.2
+	k8s.io/apimachinery => k8s.io/apimachinery v0.24.2
+	k8s.io/apiserver => k8s.io/apiserver v0.24.2
 	k8s.io/autoscaler => k8s.io/autoscaler v0.0.0-20201008123815-1d78814026aa // translates to k8s.io/autoscaler/vertical-pod-autoscaler@v0.9.0
 	k8s.io/autoscaler/vertical-pod-autoscaler => k8s.io/autoscaler/vertical-pod-autoscaler v0.9.0
```
Comment (acumino) [inline go.mod:163]: This also need to updates. Please check first commit of https://github.com/gardener/gardener-extension-provider-aws/pull/570/commits
```suggestion
k8s.io/autoscaler => k8s.io/autoscaler v0.0.0-20220531185024-cc90d57b7fe1 // translates to k8s.io/autoscaler/vertical-pod-autoscaler@v0.11.0
	k8s.io/autoscaler/vertical-pod-autoscaler => k8s.io/autoscaler/vertical-pod-autoscaler v0.11.0
```

**Comment by acumino** [comment]
Overall looks good  can you please adapt this also 
https://github.com/gardener/gardener-extension-provider-gcp/blob/0628ce4874537f60a20fe550a68b3ca48a76e84d/go.mod#L173-L174, 
Its version needs to be updated to 0.5.0

**Review by acumino — COMMENTED** [review]
/ok-to-test
/lgtm

**Review by acumino — APPROVED** [review]

**Review by dkistner — APPROVED** [review]

**Pull request merged**